kernel/ps2/src/lib.rs

```diff
@@ -105,18 +106,380 @@ pub enum HostToControllerCommand {
     // PulseOutputLineLowFor6ms = 0xF0,
 }
 
-/// Write a command to the PS/2 command port/register
-/// 
-/// Note: Devices attached to the controller should be disabled
-/// before sending commands that return data, otherwise the output buffer could get overwritten
-pub fn write_command(value: HostToControllerCommand) {
-    unsafe {
-        PS2_COMMAND_AND_STATUS_PORT.lock().write(value as u8);
+pub struct PS2Controller {
+    /// at 0x60
+    data_port: Mutex<Port<u8>>,
+    /// at 0x64
+    command_and_status_port: Mutex<Port<u8>>, //TODO: doesn't x86 have a port struct?
+}
+impl PS2Controller {
+    /// # Safety
+    /// 
+    /// Should only be called exactly once.
+    pub const unsafe fn new() -> Self {
+        Self {
+            data_port: Mutex::new(Port::new(PS2_DATA_PORT)),
+            command_and_status_port: Mutex::new(Port::new(PS2_COMMAND_AND_STATUS_PORT))
+        }
+    }
+
+    /// Get a handle to the keyboard.
+    /// 
+    /// The keyboard uses only [PS2_DATA_PORT].
+    pub fn keyboard_handle(&self) -> PS2Keyboard {
+        PS2Keyboard::new(self)
+    }
+
+    /// Get a handle to the keyboard.
+    /// 
+    /// The mouse uses both [PS2_DATA_PORT] and [PS2_COMMAND_AND_STATUS_PORT].
```

Comment:
This is a private const, so it won't show up in the docs.

```suggestion
    /// The mouse uses both the data, and command and status ports.
```